pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/RangeIndexReaderImpl.java

```diff
@@ -217,4 +272,26 @@ public void close() {
     // NOTE: DO NOT close the PinotDataBuffer here because it is tracked by the caller and might be reused later. The
     // caller is responsible of closing the PinotDataBuffer.
   }
+
+  private ImmutableRoaringBitmap getMatchesInRange(int firstRangeId, int lastRangeId) {
+    if (firstRangeId == lastRangeId) {
+      return null;
```

Comment:
If both first and last is -1, it might be no match or full match. We need to handle them separately

Reply:
Not necessarily, -1 currently means both smaller than the smallest range and larger than the largest range, so it could mean a full match or no match depending on the original query parameters. This is consistent with the current behaviour though, which appears to be incorrect.